src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/wrapper/PKCS11.java

```diff
@@ -136,6 +145,47 @@ public static void loadNative() {
     private static final Map<String,PKCS11> moduleMap =
         new HashMap<String,PKCS11>();
 
+    static boolean isKey(CK_ATTRIBUTE[] attrs) {
+        for (CK_ATTRIBUTE attr : attrs) {
+            if ((attr.type == CKA_CLASS) && (attr.getLong() == CKO_SECRET_KEY)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // This is the SunPKCS11 provider instance
+    // there can only be a single PKCS11 provider in
+    // FIPS mode.
+    private static SunPKCS11 mysunpkcs11;
+
+    private static final class InnerPKCS11 extends PKCS11 {
```

Comment:
Update this to implement `java.util.function.Consumer` like jdk11.

Reply:
updated